soap/views.py

```
@@ -0,0 +1,92 @@
+User = get_user_model()
+
+
+class GenerateSOAPView(APIView):
+    """
+    API برای تولید گزارش SOAP
+    """
+
+    permission_classes = [IsAuthenticated, IsDoctor]
+
+    @with_api_ingress(rate_limit=50, rate_window=60)
+    def post(self, request, *args, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for public function `post`

(ANN201)